src/main/java/net/snowflake/ingest/streaming/internal/BlobLatencies.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) 2023 Snowflake Computing Inc. All rights reserved.
+ */
+
+package net.snowflake.ingest.streaming.internal;
+
+import com.codahale.metrics.Timer;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import java.util.concurrent.TimeUnit;
+
+/** Latency information for a blob */
+class BlobLatencies {
+  public static final Long DEFAULT_BLOB_LATENCY = null;
+
+  private Long buildDurationMs;
+  private Long uploadDurationMs;
+
+  // flush and register duration cannot be calculated in the client sdk we pass the start time
+  // because the end time is when the request hits the server
+  private Long flushStartMs;
+  private Long registerStartMs;
+
+  public BlobLatencies() {
+    this.buildDurationMs = DEFAULT_BLOB_LATENCY;
+    this.uploadDurationMs = DEFAULT_BLOB_LATENCY;
+
+    this.flushStartMs = DEFAULT_BLOB_LATENCY;
+    this.registerStartMs = DEFAULT_BLOB_LATENCY;
```

Comment:
Are these useful? I think we want `flush duration` which I believe is covered by `uploadDurationMs`. But `registerStartMs` would be used how?